numba_dpex/core/kernel_interface/indexers.py

```diff
@@ -169,3 +220,167 @@ def __repr__(self):
             str: str representation for NdRange class.
         """
         return self.__str__()
+
+    def __eq__(self, other):
+        return (
+            self.global_range == other.global_range
+            and self.local_range == other.local_range
+        )
```

Comment:
The `other` can be any type, it should be checked to be of type `NdRange` before attempting to access attributes. Otherwise `AttributeError` would be raised.

Reply:
Addressed in af0208dde9d6763117de8789a5924243d60e910e